### PR TITLE
Add document for robot account

### PR DESCRIPTION
**Pull request by wy65701436**
Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
 In Harbor portal, enter your project, select the repository, click on the link of tag name you'd like to see its build history, the detail page will be opened. Then switch to `Build History` tab, you can see the build history information.
 
 ![build_ history](img/build_history.png)
+
+## Robot Account
+Robot accounts are accounts created by project admin that are intended for automated operations,
```
Comment (nlowe) [inline docs/user_guide.md]:
```suggestion
Robot Accounts are accounts created by project admins that are intended for automated operations. They have the following limitations:
```

File: docs/user_guide.md [hunk 2]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
 In Harbor portal, enter your project, select the repository, click on the link of tag name you'd like to see its build history, the detail page will be opened. Then switch to `Build History` tab, you can see the build history information.
 
 ![build_ history](img/build_history.png)
+
+## Robot Account
+Robot accounts are accounts created by project admin that are intended for automated operations, 
+
+1, Robot account cannot login Harbor portal.
```
Comment (nlowe) [inline docs/user_guide.md]:
```suggestion
1. Robot Accounts cannot login Harbor portal
```

File: docs/user_guide.md [hunk 3]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
 In Harbor portal, enter your project, select the repository, click on the link of tag name you'd like to see its build history, the detail page will be opened. Then switch to `Build History` tab, you can see the build history information.
 
 ![build_ history](img/build_history.png)
+
+## Robot Account
+Robot accounts are accounts created by project admin that are intended for automated operations, 
+
+1, Robot account cannot login Harbor portal.
+2, Robot account can do docker pull/push only with token.
```
Comment (nlowe) [inline docs/user_guide.md]:
```suggestion
2. Robot Accounts only perform `docker push`/`docker pull` operations with a token
```

File: docs/user_guide.md [hunk 4]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
 
 ![build_ history](img/build_history.png)
+
+## Robot Account
+Robot accounts are accounts created by project admin that are intended for automated operations, 
+
+1, Robot account cannot login Harbor portal.
+2, Robot account can do docker pull/push only with token.
+
+### Add a robot account
```
Comment (nlowe) [inline docs/user_guide.md]: Try to keep the casing consistent:

```suggestion
### Add a Robot Account
```

File: docs/user_guide.md [hunk 5]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
+If you are a project admin, you can create a robot account by clicking "New Robot Account" in the **'Robot Accounts'** tab of a project, and enter a name, a description and permission.
```
Comment (nlowe) [inline docs/user_guide.md]:
```suggestion
If you are a project admin, you can create a robot account by clicking "New Robot Account" in the `Robot Accounts` tab of a project, and enter a name, a description and permission.
```

File: docs/user_guide.md [hunk 6]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
+![add_robot_account](img/robotaccount/add_robot_account.png)
+
+![add_robot_account](img/robotaccount/add_robot_account_2.png)
+
+**NOTES:** The name will become robot$accountname that used to distinguish a robot account from harbor user.
```
Comment (nlowe) [inline docs/user_guide.md]:
```suggestion
> **NOTE:** The name will become `robot$<accountname>` and will be used to distinguish a robot account from a normal harbor user.
```

File: docs/user_guide.md [hunk 7]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
+
+![copy_robot_account_token](img/robotaccount/copy_robot_account_token.png)
+As Harbor doesn't store your account token, please make sure to copy it in the pop up dialog after creating, otherwise, there is no way to get it from Harbor.
```
Comment (nlowe) [inline docs/user_guide.md]: Is this actually the case? Or do we just not show it to the user after it's generated? I'd think it'd *have* to be stored to verify the robot during `push`/`pull` operations...

File: docs/user_guide.md (hunk 4 above)
Comment (nlowe) [reply docs/user_guide.md]: There are a few other places further down you may want to check as well

File: docs/user_guide.md (hunk 7 above)
Comment (wy65701436) [reply docs/user_guide.md]: Per security concern, Harbor doesn't store the token of robot accounts. The only way to get it is from the pop-up dialog.

**Comment by wy65701436** [comment]
@nlowe thanks for the review, I have updated the PR per your comments, please help to review the changes.

File: docs/user_guide.md (hunk 7 above)
Comment (nlowe) [reply docs/user_guide.md]: Oh I see, it's a JWT token. Makes sense to me now!

File: docs/user_guide.md [hunk 8]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
+
+![add_robot_account](img/robotaccount/add_robot_account_2.png)
+
+> **NOTE:** The name will become `robot$<accountname>` and will be used to distinguish a robot account from a normal harbor user.
+
+![copy_robot_account_token](img/robotaccount/copy_robot_account_token.png)
+As Harbor doesn't store your account token, please make sure to copy it in the pop up dialog after creating, otherwise, there is no way to get it from Harbor.
+
+### Configure duration of robot account
+If you are a system admin, you can configure the robot account token duration in days.
```
Comment (stonezdj) [inline docs/user_guide.md]: token duration -> token time to live or token expiration

File: docs/user_guide.md [hunk 9]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
+To authenticate with a Robot Account, use `docker login` as below,
+
+```
+docker login harbor.io
+Username: robot$accountname
+Password: Thepasswordgeneratedbyprojectadmin
+```
+
+### Disable a robot account
+If you are a project admin, you can disable a Robot Account by clicking "Disable Account" in the `Robot Accounts` tab of a project.
```
Comment (stonezdj) [inline docs/user_guide.md]: For project admin, you can ....

File: docs/user_guide.md [hunk 10]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
...
+![disable_robot_account](img/robotaccount/disable_delete_robot_account.png)
+
+### Delete a robot account
+If you are a project admin, you can delete a Robot Account by clicking "Delete" in the `Robot Accounts` tab of a project.
```
Comment (stonezdj) [inline docs/user_guide.md]: For project admins,

File: docs/user_guide.md [hunk 11]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
 In Harbor portal, enter your project, select the repository, click on the link of tag name you'd like to see its build history, the detail page will be opened. Then switch to `Build History` tab, you can see the build history information.
 
 ![build_ history](img/build_history.png)
+
+## Robot Account
+Robot Accounts are accounts created by project admins that are intended for automated operations. They have the following limitations:
+
+1, Robot Accounts cannot login Harbor portal
```
Comment (stonezdj) [inline docs/user_guide.md]: add a period to the end.

File: docs/user_guide.md [hunk 12]
```diff
@@ -597,3 +598,42 @@ Build history make it easy to see the contents of a container image, find the co
 In Harbor portal, enter your project, select the repository, click on the link of tag name you'd like to see its build history, the detail page will be opened. Then switch to `Build History` tab, you can see the build history information.
```
Comment (stonezdj) [inline docs/user_guide.md]: enter your project -> enter the project

Comment (wy65701436) [reply docs/user_guide.md]: it's not in this PR....

File: docs/user_guide.md (hunk 8 above)
Comment (wy65701436) [reply docs/user_guide.md]: `token duration` is an comment term when to talk about expiration. let's keep it.

File: docs/user_guide.md (hunk 9 above)
Comment (wy65701436) [reply docs/user_guide.md]: to keep the consistency of doc, can you just use `If you are a project admin`

File: docs/user_guide.md (hunk 10 above)
Comment (wy65701436) [reply docs/user_guide.md]: same

**Pull request merged**